### PR TITLE
Fix fluent interface inconsistency

### PR DESCRIPTION
**Pull request by im-denisenko**
- All methods that supposed to be fluent now return instance from here they were called.
- Some tests were added.
- `@return` docblock updated across entire library. Finishes work started in #774.
- `@param` and `@return` realigned using php-cs-fixer.

### CODE REVIEW

File: lib/Elastica/Bulk/Action/AbstractDocument.php [hunk 1]
```diff
@@ -114,9 +115,9 @@ public function getData()
     abstract protected function _getMetadata(AbstractUpdateAction $source);
 
     /**
-     * @param  \Elastica\Document|\Elastica\Script    $data
-     * @param  string                                 $opType
-     * @return \Elastica\Bulk\Action\AbstractDocument
+     * @param  \Elastica\Document|\Elastica\Script $data
+     * @param  string                              $opType
+     * @return static
```
Comment (ruflin) [inline lib/Elastica/Bulk/Action/AbstractDocument.php]: I like that :-)

File: lib/Elastica/Connection.php [hunk 2]
```diff
@@ -254,9 +256,10 @@ public function getConfig($key = '')
     }
 
     /**
-     * @param  \Elastica\Connection|array $params Params to create a connection
      * @throws Exception\InvalidException
-     * @return \Elastica\Connection
+     *
+     * @param  \Elastica\Connection|array $params Params to create a connection
+     * @return self
```
Comment (ruflin) [inline lib/Elastica/Connection.php]: :+1:

File: lib/Elastica/Query/Fuzzy.php [hunk 3]
```diff
@@ -64,7 +63,7 @@ public function setFieldOption($param, $value)
         $keyArray = array_keys($params);
         $params[$keyArray[0]][$param] = $value;
 
-        return $this->setparam($keyArray[0], $params[$keyArray[0]]);
```
Comment (ruflin) [inline lib/Elastica/Query/Fuzzy.php]: Good spot ...

**Comment by ruflin** [comment]
Awesome. Thanks.

**Comment by ruflin** [comment]
@im-denisenko BTW: I think you will like this: https://scrutinizer-ci.com/g/ruflin/Elastica/?branch=master

**Comment by im-denisenko** [comment]
I quickly looked at reported issues, and yeah, it's awesome :)

Though, many of them seems to be false-positive due to incorrect phpdoc comments (this tool gets variable types from docblocks, as I can guess)

Speaking of that, I see 1527 warnings every time I trying to run phpdoc. I already wanted to fix that, so now there yet another reason to do this.

**Comment by ruflin** [comment]
:-)

**Pull request merged**